processing/src/main/java/io/druid/query/groupby/having/HavingSpecMetricComparator.java

```diff
@@ -0,0 +1,53 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.query.groupby.having;
+
+import com.google.common.primitives.Floats;
+import com.google.common.primitives.Longs;
+import io.druid.data.input.Row;
+
+import java.util.regex.Pattern;
+
+/**
+ */
+class HavingSpecMetricComparator
+{
+  private static final Pattern LONG_PAT = Pattern.compile("[-|+]?\\d+");
```

Comment:
Can you make this package protected so you can test it explicitly?

specifically I would like to see a test for matching against `3456.3` or similar